analyses/snv-callers/vaf_cutoff_experiment.Rmd

```diff
@@ -0,0 +1,284 @@
+
+  # This is to remove redundancy as upper correlation matrix == lower
+  cor_res[upper.tri(cor_res, diag = TRUE)] <- NA
+
+  # Format into long format data.frame
+  cor_df <- reshape2::melt(cor_res, na.rm = TRUE, value.name = "cor") %>%
+    dplyr::mutate(
+      compare = paste0(Var1, "-", Var2),
+      cutoff
+    )
+
+  # Return a data.frame ready for being added to the bigger data.frame for plotting.
+  return(cor_df)
+}
+```
+
+Calculate variance for a sample for a given caller amongst the different VAF cutoffs.
+
+```{r}
+var_by_sample <- function(sample) {
+  # For a given sample ID, isolate the TMB statistics and calculate the variance.
+  #
+  # Args:
+  #   sample: Tumor_Sample_Barcode of the sample you would like the variance for.
+  #
+  # Returns:
+  #   Variance for that sample for each caller.
+
+  sample_df <- tmb_filter_long %>%
+    dplyr::filter(Tumor_Sample_Barcode == sample) %>%
+    dplyr::select(-Tumor_Sample_Barcode, -vaf_cutoff)
+
+  # Do correlations between callers
+  var_by_caller <- data.frame(
+    t(apply(sample_df, 2, var)),
+    sample
+  )
+
+  return(var_by_caller)
+}
+```
+
+## Run the data with different VAF filter cutoffs
+
+We need to re-calculate TMB statistics with various VAF filter cutoffs. 
+Run this script to do it. 
+
+```{r include=FALSE}
+system("bash scripts/run_caller_evals-vaf_filter_exp.sh")
+```
+
+## Import data 
+
+Get the lists of each type of file. 
+
+```{r}
+# Make a list of the vaf filter experiment tmb files
+filter_tmb_files <- list.files(file.path(base_results_dir, "vaf_filter"),
+  pattern = "_tmb.rds$", recursive = TRUE,
+  full.names = TRUE
+)
+```
+
+Take a look at the TMBs from different VAF filter cutoffs. 
+
+```{r}
+filter_tmb_files
+```
+
+Convert the VAF cutoffs into their own vector. 
+
+```{r}
+vaf_filter <- stringr::word(filter_tmb_files, sep = "/", 3)
+vaf_filter <- as.numeric(gsub("cutoff_", "", vaf_filter))
+vaf_filter
+```
+
+Get caller information as a vector. 
+
+```{r}
+caller_filter <- stringr::word(filter_tmb_files, sep = "/", 4)
+caller_filter <- gsub("_tmb.rds", "", caller_filter)
+caller_filter
+```
+
+Read in the data. 
+Name each with their caller and VAF filter cutoff. 
+
+```{r include=FALSE}
+filter_tmb_list <- lapply(filter_tmb_files, function(file_name) {
+  readr::read_rds(file_name) %>% dplyr::ungroup()
+})
+names(filter_tmb_list) <- paste0(caller_filter, "_", vaf_filter)
+```
+
+Turn this into a data.frame
+
+```{r}
+tmb_filter_df <- dplyr::bind_rows(filter_tmb_list, .id = "NAME") %>%
+  dplyr::mutate(
+    caller = stringr::word(NAME, sep = "_", 1),
+    vaf_cutoff = as.numeric(stringr::word(NAME, sep = "_", 2))
+  )
+```
+
+Make this into a long form data.frame for plotting purposes. 
+
+```{r}
+tmb_filter_long <- tmb_filter_df %>%
+  dplyr::distinct(Tumor_Sample_Barcode, caller, tmb, vaf_cutoff) %>%
+  tidyr::spread(caller, tmb)
+```
+
+## Make the TMB correlation plots
+
+```{r}
+lapply(unique(vaf_filter), cor_matrix_plot)
```

Comment:
Out of curiosity, would further increasing the VAF cutoff continue to give a better correlation or is there a cutoff that you found to be optimal based on this analysis?

Reply:
After a certain cutoff, there are not many mutations left, but this range of VAF filter cutoffs seemed to give us a good enough idea of what was going on.